src/game.h

```diff
@@ -1027,6 +1029,10 @@ class game
         int turnssincelastmon = 0; // needed for auto run mode
 
         weather_manager weather;
+        std::vector<std::pair<generic_requirement_type_id, generic_event_type_id>> generic_events_vector;
+        std::vector<std::pair<time_point, generic_event_type_id>> queued_generic_events;
+        std::map<std::string, bool> generic_variable_map;
```

Comment:
If you're going to create a collection of variable accessible to json for arbitrary purposes then we might want to make them `cata_variant`, not just `bool`, so they can hold all sorts of values.

Also, have you thought about world-level variables vs character-level variables?  We have problems already in the game with some things not working logically when you play a new character in the same world.  It would be good not to worsen this issue (in the hopes that someday it can be made better), so I think we might want both varieties of variable, and they need to be saved in different places.

You don't need to add both in this PR, but please think carefully about what you're currently using them for and which type of variable they should be, and what terminology we should be using to distinguish the two.

Reply:
So I looked into your suggestion about using cata_variant but it feels like it would add a ton of complexity to the json loading and to attempts to use it in comparisons.  I'm not sure if I'm missing an easy way to handle it all though as I'm not super familiar with cata_variant.

At the moment my plan is to replace the bool with an int.  An int could be changed into some of the cata_varaint types if needed but I think adds a ton of power without making the json too complicated.  The other thing it does it open up using generic_variables as values to other preconditions/operations.  So you could set up a pair like as long as the temperature is below var1 add pain equal to var2.

I hadn't thought about character versus world events at all and its a bit of a doozy, though good to think about.  At the moment everything is character level I think, though I might need to double check the save code.